### ISSUE TITLE
Sidebar Search Issue

### ISSUE THREAD

**Issue by altworkz** [retypeapp/retype#646]
Is it by design that the items with `index.yml` cannot be searched in the sidebar? The structure is:

> folder1
> --index.md (label: data configuration)
> folder2
> --index.yml (label: data manipulation)
> folder3
> --index.md (label: data structures)

The header title from `folder2` is not showing in the sidebar search when you search for the word "data."

But, when the structure is:

> folder4
> --folder4.1
> ----folder4.1.1
> ------index.yml (label: data management)

The header appears in the search bar. I'm wondering if the header from `folder4` is just showing because there are other `.md` files inside the `folder4.1.1` with `label: data something`.

**Comment by geoffreymcgill** [comment]
Hi @altworkz. Any chance you could publish your project or a stripped down project to GitHub demonstrating how you have this all configured? Or even just attaching a simplified project structure as a .zip file to this issue?

I think the issue here is that [`label`](https://retype.com/configuration/page/#label) should only be used to set the label of the node in the left tree navigation and is not added to the search index, as the `label` is not actually used for any content of the page if the page also has a primary heading.

The most basic `.md` page would be:

```md
# My Page Title
```

You can change the left tree nav label by setting the `label` page config:

```md
---
label: Custom label
---
# My Page Title
```

If I add a `label`, the left tree nav will now say `Custom label`, but that value is not added to the search index. Within the search, the term `My Page Title` will be result.

In generally, I would recommend not using the `label` or the `title` config. The best approach is to set the primary heading of each page with a proper heading, such as `# My Page Title`. Then the left tree nav label and breadcrumb label are generated automatically from the page `H1` heading. 

Retype will use the `label` config as the page heading when no page heading is available. This might be what is happening in your `folder2` and `folder4.1.1` scenarios.

When a page `# H1 heading` is available, Retype will add that value to the search index, even if a `label` is also set.

If only a `label` is available (no `# H1 heading`), then Retype will use the `label` for the tree nav label, the page heading, and  also add that value to the search index.

Hope this helps.

**Comment by altworkz** [comment]
Hello @geoffreymcgill, I'll be attaching here the simplified project structure. 

In this setup:
![image](https://github.com/retypeapp/retype/assets/133190677/2b1ae41a-60aa-4bb0-8a26-6ef70e57f106)

We expected that 2 data manipulation will still show when "data" is searched.
![image](https://github.com/retypeapp/retype/assets/133190677/4fb78410-05f7-4952-8a9b-23fbce880d6c)

[label-issue.zip](https://github.com/retypeapp/retype/files/13042562/label-issue.zip)

**Comment by geoffreymcgill** [comment]
> We expected that 2 data manipulation will still show when "data" is searched.

Thanks. I have reproduced. 

Looks like the filter is only filtering on the content files, not the folder names.

We will investigate and try to improve the functionality.

**Comment by geoffreymcgill** [comment]
This sidebar filter issue has been fixed and the fix will be included in the upcoming v3.6.0 release. 

<img width="291" alt="Screen Shot 2023-10-20 at 4 15 05 PM" src="https://github.com/retypeapp/retype/assets/62210/b87a6086-0d1b-4fb7-b996-6440f4e99914">


https://github.com/retypeapp/retype/assets/62210/2cc600e0-5faa-4404-b89f-b36dfcac0e33
